Extensions/ExternalTfs/Src/Tasks/DownloadExternalBuildArtifacts/download.ts

```diff
@@ -55,27 +55,44 @@ function publishEvent(feature, properties: any): void {
 
 async function main(): Promise<void> {
     var promise = new Promise<void>(async (resolve, reject) => {
-        var projectId = tl.getInput("project", true);
-        var definitionId = tl.getInput("definition", true);
-        var buildId = tl.getInput("version", true);
-        var itemPattern = tl.getInput("itemPattern", true);
-        var downloadPath = tl.getInput("downloadPath", true);
-
-        var endpointUrl : any;
-        var username : any;
-        var accessToken : any;
-
-        if(isAdoServiceConnectionSet()) {
-            var connection = tl.getInput("azureDevOpsServiceConnection", true);
-            endpointUrl = tl.getEndpointUrl(connection, false);
+        // Read inputs from the user
```

Comment:
Maybe instead of getting all possible inputs here we may first check type (ado or not) and based on this information retrieve only what we need?